### PR TITLE
361 cyclocomp linter

### LINKED ISSUE #361: Linter for function length
It looks like there's no existing linter for function length (number of lines, or number of lines of code).  Would that be a valuable addition?

I ask because I've been asked to code review a package that other people wrote, and there are several extremely long functions, and it would be nice to get a linter to report these facts for me. =)

Obviously I am aware that there's no agreed-upon max length for functions, that number of lines of code is a poor indicator of complexity, etc. - would like to get something that at least gets in the right ballpark, though.

### PR DESCRIPTION
**Pull request by fabian-s**
adds (function-wise) linting for cyclomatic complexity.  
computes cyclomatic complexity for each entry in `source_files$expressions`. 

incurs a dependency on `cyclocomp`.  
closes #361.

Not sure if the generated message is too verbose for your taste, I figured it's ok since this will only be generated at most once for any top-level expression in each file.

### CODE REVIEW

File: R/cyclocomp_linter.R [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+      cyclocomp::cyclocomp(parse(text = source_file$content))
+    )
+    if (inherits(complexity, "try-error")) return(NULL)
+    if (complexity <= complexity_limit) return(NULL)
+    Lint(
+      filename = source_file[["filename"]],
+      line_number = source_file[["line"]][1],
+      column_number = source_file[["column"]][1],
+      type = "style",
+      message = paste0(
```
Comment (jimhester) [inline R/cyclocomp_linter.R]: I think this is too long, maybe

"functions should have cyclomatic complexity of less than", complexity_limit, "."

**Comment by jimhester** [comment]
I think this looks good, thanks! 
Can you please add a bullet to NEWS? It should briefly [describe the change](https://style.tidyverse.org/news.html) and end with `(@yourname, #issuenumber)`.

**Comment by jimhester** [comment]
It looks like you have a few test failures, once those are fixed and we have a clean build this can be merged.

**Comment by fabian-s** [comment]
> It looks like you have a few test failures, once those are fixed and we have a clean build this can be merged.

sorry about that, forgot to update the tests to the new message... :see_no_evil:

**Comment by jimhester** [comment]
Great work, this package is now better thanks to you!

Thanks!! You are da bomb!

**Pull request merged**